server/index.js

```diff
@@ -51,6 +51,86 @@ app.get('/api/products/:productId', (req, res, next) => {
     .catch(err => next(err));
 });
 
+app.get('/api/cart', (req, res, next) => {
+  if (!req.session.cartId) {
+    res.status(200).json({});
+  }
+  const sql = `
+              select "c"."cartItemId",
+                     "c"."price",
+                     "p"."productId",
+                     "p"."image",
+                     "p"."name",
+                     "p"."shortDescription"
+                from "cartItems" as "c"
+                join "products" as "p" using ("productId")
+              where  "c"."cartId" = $1
+              `;
+  const values = [req.session.cartId];
+  db.query(sql, values)
+    .then(result => res.status(200).json(result.rows))
+    .catch(err => next(err));
+});
+
+app.post('/api/cart', (req, res, next) => {
+  const productId = parseInt(req.body.productId);
+  if (isNaN(productId) || productId < 0) {
+    next(new ClientError('productId must be positive integer', 400));
+  }
+  const sql = `
+              select "price"
+              from "products"
+              where "productId" = $1
+              `;
+  const values = [productId];
+  db.query(sql, values)
+    .then(result => {
+      if (result.rows.length === 0) {
+        next(new ClientError(`Cannot find product with productId of ${productId}`, 404));
```

Comment:
You should `throw` the client error instead of passing it to `next()`. Calling `next` will not stop the promise chain.